app/actions/app_apply_manifest.rb

```diff
@@ -157,21 +162,41 @@ def create_service_bindings(manifest_service_bindings_message, app)
             message: binding_message)
 
           begin
-            result = action.bind(binding, parameters: binding_message.parameters)
-            if result[:async]
-              raise ServiceBrokerRespondedAsyncWhenNotAllowed
-            end
-          rescue ServiceBrokerRespondedAsyncWhenNotAllowed,
-                 V3::ServiceBindingCreate::BindingNotRetrievable
-
-            raise ServiceBrokerRespondedAsyncWhenNotAllowed.new('The service broker responded asynchronously, but async bindings are not supported.')
+            binding = action.bind(binding, parameters: binding_message.parameters, accepts_incomplete: false)
+          rescue ServiceBrokerRespondedAsyncWhenNotAllowed
```

Comment:
`action.bind` is not raising `ServiceBrokerRespondedAsyncWhenNotAllowed`, or?

Reply:
We've replaced this error with the error we actually see get raised when testing with the OverviewBroker set to async only.

We're not sure if we need to handle this case:
```
result = action.bind(binding, parameters: binding_message.parameters)
if result[:async]
    raise ServiceBrokerRespondedAsyncWhenNotAllowed
end
```

According to the [OSBAPI spec](https://github.com/openservicebrokerapi/servicebroker/blob/v2.17/spec.md#asynchronous-operations), brokers should reject the request if `accepts_incomplete` is not set to true and they can not handle the request asynchronously.

Reply:
I'm unsure about this. On the one hand the OSBAPI spec is pretty clear (MUST) on the other hand it might be worth to keep this check as it has been there in the past...

What about moving the check [into the action](https://github.com/cloudfoundry/cloud_controller_ng/blob/3dcc660170c86c581b0d26ee4290028f38cbbd75/app/actions/v3/service_binding_create.rb#L27)? This would mean introducing a new error, e.g. `AsyncNotAllowed`, that needs to be handled in `AppApplyManifest`.

Two other remarks:
- The action (`ServiceBindingCreate`) also raises a `BindingNotRetrievable` error, which should be handled here as well.
- If the check is either removed or done within the action, it's not necessary to assign the binding variable.

Reply:
Can you clarify why `BindingNotRetrievable` needs to be handled?  My understanding that if that is raised, it means that the Service Broker doesn't support fetching/polling for the Binding status, and so we should error out (i.e. keep the current behavior for those cases)

Reply:
@sethboyles You are right. Changing the error message in that case (from "The service broker responded asynchronously, but async bindings are not supported." to "The broker responded asynchronously but does not support fetching binding data") is correct.

Reply:
@philippthun thanks for the feedback.  Regarding the `result[:async]`, I opted to readd it to AppApplyManifest, mostly because I don't have time at the moment to dig into how other parts of the code that use the `ServiceBindingCreate` action might be affected by that change.  But by keeping the check (and erroring out like before) it will hopefully be clearer when something goes wrong to users than the undefined behavior we would get if we removed the check.